python_modules/libraries/dagster-dg/dagster_dg/cli/component.py

```diff
@@ -246,8 +245,7 @@ def component_list_command(context: click.Context, **global_options: object) ->
     """List Dagster component instances defined in the current code location."""
     cli_config = normalize_cli_config(global_options, context)
     dg_context = DgContext.from_config_file_discovery_and_cli_config(Path.cwd(), cli_config)
-    if not dg_context.is_code_location:
-        exit_with_error("This command must be run inside a Dagster code location directory.")
+    dg_context.validate_code_location_command_environment()
```

Comment:
seems like this validation should happen in the factory method so people don't forget to add it in future callsites

Reply:
What exactly do you mean by "the factory method"? `@component_group.command`? That's part of `click`. And even if we were to override it, a difficulty is that different commands have different requirements-- some need a component lib but not code location, some just need `dagster-components` available, some need a code location. That's why there are multiple `DgContext.validate_*` methods.

I think the extra abstraction required to make the correct check automatic is not worth it, especially since people will typically copy an existing command as the template for their new one.

Reply:
All I mean is that 

```python
    dg_context = DgContext.from_config_file_discovery_and_cli_config(Path.cwd(), cli_config)
    dg_context.validate_code_location_command_environment()
```
should be a single function that creates the dg context and validates the code location

Reply:
`dg_context = DgContext.for_code_location_env(...)`

Reply:
Good call, updated